### PR TITLE
fix: when called checker:get_target_status got err, we should log it~

### PR DESCRIPTION
**Pull request by yongboy**
### What this PR does / why we need it:

For issues #3423

If we call `checker:get_target_status` got err, we should log it, helping find some hidden mistakes.

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Review by spacewander — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Comment by tokers** [comment]
LGTM but the CI failed @yongboy

File: apisix/balancer.lua [hunk 1]
```diff
@@ -55,10 +55,12 @@ local function fetch_health_nodes(upstream, checker)
     local port = upstream.checks and upstream.checks.active and upstream.checks.active.port
     local up_nodes = core.table.new(0, #nodes)
     for _, node in ipairs(nodes) do
-        local ok = checker:get_target_status(node.host, port or node.port, host)
+        local ok, err = checker:get_target_status(node.host, port or node.port, host)
         if ok then
             -- TODO filter with metadata
             up_nodes[node.host .. ":" .. node.port] = node.weight
+        elseif err then
+            core.log.error("failed to get health check target status: ", err)
```
Comment (spacewander) [inline apisix/balancer.lua:63]: Just be captious, it would be better to log down the arguments `node.host, port or node.port, host` when error occurs.

**Comment by spacewander** [comment]
The CI failed because of a hidden bug: https://github.com/apache/apisix/pull/3427
Thanks @yongboy for discovering it!

**Comment by spacewander** [comment]
Please merge master to make CI pass.

File: apisix/balancer.lua (hunk 1 above)
Comment (yongboy) [reply apisix/balancer.lua:63]: Copy ~

File: apisix/balancer.lua [hunk 2]
```diff
@@ -55,10 +55,12 @@ local function fetch_health_nodes(upstream, checker)
     local port = upstream.checks and upstream.checks.active and upstream.checks.active.port
     local up_nodes = core.table.new(0, #nodes)
     for _, node in ipairs(nodes) do
-        local ok = checker:get_target_status(node.host, port or node.port, host)
+        local ok, err = checker:get_target_status(node.host, port or node.port, host)
         if ok then
             -- TODO filter with metadata
             up_nodes[node.host .. ":" .. node.port] = node.weight
+        elseif err then
+            core.log.error("failed to get health check target status, addr: ", node.host, ":", port or node.port, ", host: ", host, ", err: ", err)
```
Comment (spacewander) [inline apisix/balancer.lua:63]: The line is too long.
```
Checking apisix/balancer.lua                      1 warning

    apisix/balancer.lua:63:101: line is too long (147 > 100)

Total: 1 warning / 0 errors in 130 files
Makefile:77: recipe for target 'lint' failed
```

**Comment by yongboy** [comment]
Now calling `checker:get_target_status` get error, the log example as below:

```bash
2021/01/27 11:34:52 [error] 8#8: *2315 [lua] balancer.lua:73: fetch_health_nodes(): failed to get health check target status, addr: 172.19.5.113:8000, host: nil, err: target not found while connecting to upstream, client: 172.19.0.1, server: , request: "GET /web.txt HTTP/1.1", host: "web1.lvh.me"
```

File: apisix/balancer.lua (hunk 2 above)
Comment (yongboy) [reply apisix/balancer.lua]: Fixed :))

**Pull request merged**